### PR TITLE
Remove support for servicemanager and eventmanager v2

### PR DESCRIPTION
**Pull request by snapshotpl**
zend-servicemanager v2 support is dropped in v3 so we can remove tone of code...

### CODE REVIEW

**Comment by snapshotpl** [comment]
... and also zend-eventmanager v2.

File: src/Service/ServiceListenerFactory.php [hunk 1]
```diff
@@ -96,20 +97,6 @@ class ServiceListenerFactory implements FactoryInterface
...
-     * Constructor
-     *
-     * When executed under zend-servicemanager v3, injects additional aliases
-     * to ensure backwards compatibility.
-     */
-    public function __construct()
-    {
-        $r = new ReflectionClass(ServiceLocatorInterface::class);
-        if ($r->hasMethod('build')) {
-            $this->injectV3Aliases();
```
Comment (Xerkus) [inline src/Service/ServiceListenerFactory.php:108]: breaking change, high chance someone still uses those compatibility aliases

File: src/Service/ServiceListenerFactory.php [hunk 2]
```diff
@@ -263,22 +239,4 @@ private function validatePluginManagerOptions($options, $name)
...
-     * Inject additional aliases for zend-servicemanager v3 usage
-     *
-     * If the constructor detects that we're operating under zend-servicemanager v3,
-     * this method injects additional aliases to ensure that common services
-     * can be retrieved using both Titlecase and lowercase, and will get the
-     * same instances.
-     *
-     * @return void
-     */
-    private function injectV3Aliases()
```
Comment (Xerkus) [inline src/Service/ServiceListenerFactory.php:277]: breaking change, high chance someone still uses those compatibility aliases

**Review by Xerkus — CHANGES_REQUESTED** [review]
Great stuff, however there are couple BC concerns.

File: src/Service/ServiceListenerFactory.php (hunk 1 above)
Comment (snapshotpl) [reply src/Service/ServiceListenerFactory.php:108]: @Xerkus you have right. That's why I add aliases into `$defaultServiceConfig` property (above)

**Review by Xerkus — APPROVED** [review]

File: src/Service/ServiceListenerFactory.php (hunk 1 above)
Comment (Xerkus) [reply src/Service/ServiceListenerFactory.php:108]: Indeed

**Pull request merged**